### DISCUSSION TITLE
negative potentials on atoms of positively charged residues

### DISCUSSION THREAD

**Discussion by deekshatcb** [Electrostatics/apbs#181] [Q&A]
I want to calculate potential on specific atoms of some residues in a protein. From the .dx file (using the interpolate function) I was extracting the potentials on the grid points nearest to the atoms of interest, some of which happens to be NZ atoms of lysine. I am getting negative potentials. Is it normal? or am I doing something wrong?

**Comment by sobolevnrm** [comment]
Can you please say more about how you're performing the calculation?  Is this for a single calculation or for the difference of two calculations (e.g., for solvation energy and reaction potential)?

Note that you can get the value of the potential at an atom center using [write atompot](https://apbs.readthedocs.io/en/latest/using/input/old/elec/write.html).

**Reply by deekshatcb** [reply]
I am attaching the apbs input file for more details. This is for single calculation. I want to know the potential values at NZ atoms of some specific lysine (131, 215, 277).
********************************************************************
read
    mol pqr 1e19.pqr
end
elec 
    mg-auto
    dime 161 161 161
    cglen 103.7935 94.5506 103.2665
    fglen 81.0550 75.6180 80.7450
    cgcent mol 1
    fgcent mol 1
    mol 1
    lpbe
    bcfl sdh
    pdie 2.0000
    sdie 78.5400
    srfm smol
    chgm spl2
    sdens 10.00
    srad 1.40
    swin 0.30
    temp 298.15
    calcenergy total
    calcforce no
    write pot dx 1e19.pqr
end
print elecEnergy 1 end
quit

**Reply by deekshatcb** [reply]
I used the atompot option (which gave me positive potential values for the lysine). Now, for another PDB I am getting positive potential for OD1 atom of Aspartate.

**Comment by sobolevnrm** [comment]
It is unlikely that a single calculation will give you meaningful results because of the "self-energy" terms due to the discretized atomic charge distributions on a grid:  in the limit of infinitesimal grid space, all of these energies (and associated potentials) should become infinite.  With finite grid spacings, you can still get very large energies with very large errors.

The best way to calculate atomic potentials through three separate calculations:

1. Calculate atomic potentials φ<sub>1</sub> with an inhomogeneous dielectric coefficient:  ε<sub>u</sub> in the solute and <sub>v</sub> in the solvent.
2. Calculate atomic potentials φ<sub>2</sub> with a homogeneous dielectric coefficient:  ε<sub>u</sub> everywhere
3. Use Coulomb's law with a dielectric of <sub>u</sub> to calculate the potential φ<sub>3</sub> on the atom of interest due to all other atoms in the molecule.

The correct potential at your atom should then be φ<sub>1</sub> - φ<sub>2</sub> + φ<sub>3</sub>.
